repose-aggregator/components/filters/openstack-identity-v3-filter/src/main/scala/org/openrepose/filters/openstackidentityv3/utilities/OpenStackIdentityV3API.scala

```diff
@@ -161,10 +168,33 @@ class OpenStackIdentityV3API(config: OpenstackIdentityV3Config, datastore: Datas
             // because we care to match on the status code of the response, if anything was set.
             validateTokenResponse.map(response => response.getStatus) match {
               case Some(statusCode) if statusCode == HttpServletResponse.SC_OK =>
-                val subjectTokenObject = jsonStringToObject[AuthResponse](inputStreamToString(validateTokenResponse.get.getData)).token
+                val json = Json.parse(inputStreamToString(validateTokenResponse.get.getData))
 
-                val expiration = new DateTime(subjectTokenObject.expires_at)
-                val identityTtl = safeLongToInt(expiration.getMillis - DateTime.now.getMillis)
+                val subjectTokenObject = ValidToken(
```

Comment:
Why are we doing this instead of just working with the json object from play directly?